README.md

```diff
@@ -18,6 +18,122 @@ To install directly from this branch:
 
 - [Documentation](https://prometheus-api-client-python.readthedocs.io/en/master/source/prometheus_api_client.html)
 
+## Getting Started
+
+### Usage
+[Prometheus](https://prometheus.io/), a Cloud Native Computing Foundation project, is a systems and service monitoring system. It collects metrics (time series data) from configured targets at given intervals, evaluates rule expressions, displays the results, and can trigger alerts if some condition is observed to be true. The raw time series data obtained from a Prometheus host can sometimes be hard to interpret. To help better understand these metrics we have created a Python wrapper for the Prometheus http api for easier metrics processing and analysis.
+
+The `prometheus-api-client` library consists of multiple modules which assist in connecting to a Prometheus host, fetching the required metrics and performing various aggregation operations on the time series data.
+
+#### Connecting and Collecting Metrics from a Prometheus host
+The `PrometheusConnect` module of the library can be used to connect to a Prometheus host. This module is essentially a class created for the collection of metrics from a Prometheus host. It stores the following connection parameters:
+
+-   **url** - (str) url for the prometheus host
+-   **headers** – (dict) A dictionary of http headers to be used to communicate with the host. Example: {“Authorization”: “bearer my_oauth_token_to_the_host”}
+-   **disable_ssl** – (bool) If set to True, will disable ssl certificate verification for the http requests made to the prometheus host
+
+```ruby
+from prometheus_api_client import PrometheusConnect
+prom = PrometheusConnect(url ="<prometheus-host>", disable_ssl=True)
+
+# Get the list of all the metrics that the Prometheus host scrapes
+prom.all_metrics()
+```
+
+You can also fetch the time series data for a specific metric using custom queries as follows:
+
+```ruby
+prom = PrometheusConnect()
+my_label_config = {'cluster': 'my_cluster_id', 'label_2': 'label_2_value'}
+prom.get_current_metric_value(metric_name='up', label_config=my_label_config)
+
+# Here, we are fetching the values of a particular metric name
+prom.custom_query(query="prometheus_http_requests_total")
+
+# Now, lets try to fetch the `sum` of the metrics
+prom.custom_query(query="sum(prometheus_http_requests_total)")
+```
+
+We can also use custom queries for fetching the metric data in a specific time interval. For example, let's try to fetch the past 2 days of data for a particular metric in chunks of 1 day:
+
+```ruby
+start_time = parse_datetime("2d")
```

Comment:
Can you show the imports for `parse_datetime`?

Reply:
also for `timedelta`

Reply:
added :+1: